src/pebble.py

```diff
@@ -50,14 +50,28 @@ def __init__(self, charm_state: CharmState):
         """
         self._charm_state = charm_state
 
-    def replan(self, container: ops.model.Container) -> None:
+    def replan(self, container: ops.model.Container, restart: bool = False) -> None:
         """Replan the pebble service.
 
         Args:
             container: Charm container.
+            restart: Force restart of the container. Default: False.
         """
+        logger.debug("Replaning the container")
         container.add_layer(SYNAPSE_CONTAINER_NAME, self._pebble_layer, combine=True)
         container.replan()
+        if restart:
+            container.restart(SYNAPSE_SERVICE_NAME)
```

Comment:
In the case where a replan is needed and restarts the service *and* `restart` is True, this will restart the service twice in succession -- probably not what you want. See comment below too. I think trying to roll up a "replan maybe restart" into one function here is a bit confusing.

Reply:
Good point, thanks @benhoyt ! @amandahla fyi in case you missed it

Reply:
Thanks, I changed it for a restart now :)